src/main.rs

```diff
@@ -33,44 +55,250 @@ fn main() -> Result<()> {
     emulate_zx(display::create!(peripherals)?, display::color_conv)
 }
 
+
+/// This configuration is picked up at compile time by `build.rs` from the
+/// file `cfg.toml`.
+#[toml_cfg::toml_config]
+pub struct Config {
+    #[default("Wokwi-GUEST")]
+    wifi_ssid: &'static str,
+    #[default("")]
+    wifi_psk: &'static str,
+}
+
+
+
+#[allow(dead_code)]
+fn wifi(
+    netif_stack: Arc<EspNetifStack>,
+    sys_loop_stack: Arc<EspSysLoopStack>,
+    default_nvs: Arc<EspDefaultNvs>,
+) -> anyhow::Result<Box<EspWifi>> {
+    let app_config = CONFIG;
+    let mut wifi = Box::new(EspWifi::new(netif_stack, sys_loop_stack, default_nvs)?);
+
+    wifi.set_configuration(&Configuration::Client(ClientConfiguration {
+        ssid: app_config.wifi_ssid.into(),
+        password: app_config.wifi_psk.into(),
+        auth_method: AuthMethod::None,
```

Comment:
Does using `AuthMethod::None` work on WPA2 networks? Maybe we could do some checks [like in Ferrous System's repo](https://github.com/ferrous-systems/espressif-trainings/pull/54/files)